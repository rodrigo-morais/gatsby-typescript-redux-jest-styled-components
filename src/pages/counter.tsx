import React from 'react'
import { connect } from 'react-redux'

export const CounterPage = ({increment, decrement}: {increment: () => void, decrement: () => void}) => (
  <div>
    <h1>Here you can count up or down</h1>
    <button onClick={increment}>Up</button>
    <br/>
    <br/>
    <button onClick={decrement}>Down</button>
  </div>
)

const mapDispatchToProps = (dispatch: (action: any) => void) => {
  return {
    decrement: () => dispatch({ type: 'DECREMENT' }),
    increment: () => dispatch({ type: 'INCREMENT' })
  }
}

export default connect(null, mapDispatchToProps)(CounterPage)
