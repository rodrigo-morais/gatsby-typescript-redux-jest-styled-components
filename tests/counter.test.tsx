import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { CounterPage } from '../src/pages/counter'

const increment = jest.fn()
const decrement = jest.fn()
const props = { increment, decrement }
const component = (<CounterPage {...props} />)


describe('Counter', () => {
  describe('Rendering', () => {
    it('renders according to design', () => {
      const tree = renderer.create(component).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('Events', () => {
    const wrapper = shallow(component)

    beforeEach(() => {
      increment.mockReset()
      decrement.mockReset()
    })

    describe('when click on up button', () => {
      it('calls increment function', () => {
        wrapper.find('button').first().simulate('click')

        expect(increment).toBeCalled()
      })
    })

    describe('when click on down button', () => {
      it('calls decrement function', () => {
        wrapper.find('button').last().simulate('click')

        expect(decrement).toBeCalled()
      })
    })
  })
})
