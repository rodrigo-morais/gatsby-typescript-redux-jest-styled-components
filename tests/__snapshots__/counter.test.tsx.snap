// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Rendering renders according to design 1`] = `
<div>
  <h1>
    Here you can count up or down
  </h1>
  <button
    onClick={[MockFunction]}
  >
    Up
  </button>
  <br />
  <br />
  <button
    onClick={[MockFunction]}
  >
    Down
  </button>
</div>
`;
